tests/jballerina-semtype-test/src/test/java/io/ballerina/test/SemTypeTest.java

```diff
@@ -0,0 +1,138 @@
+        for (File file : Objects.requireNonNull(dir.listFiles())) {
+            String fileName = file.getName();
+            if (fileName.contains(FAILING_FILE) || fileName.contains(DISABLED_FILE)) {
+                continue;
+            }
+            files.add(file.getAbsolutePath());
+        }
+        return files.toArray();
+    }
+
+    @Test(dataProvider = "filePathProvider")
+    public void testSubTypeRelationship(String filePath) throws IOException {
+        Set<String> actualRelations = actualSubTypeRelations(filePath);
+        Set<String> expectedRelations = expectedSubTypeRelations(filePath);
+        Assert.assertEquals(actualRelations.size(), expectedRelations.size());
+        actualRelations.removeAll(expectedRelations);
+        Assert.assertTrue(actualRelations.isEmpty());
+    }
+
+    private Set<String> actualSubTypeRelations(String filePath) {
+        CompileResult compileResult = BCompileUtil.compile(filePath);
+        BLangPackage bLangPackage = (BLangPackage) compileResult.getAST();
+        List<BTypeDefinitionSymbol> bTypeDefinitionSymbols = new ArrayList<>();
+        for (Scope.ScopeEntry value : bLangPackage.symbol.scope.entries.values()) {
+            BSymbol bSymbol = value.symbol;
+            if (bSymbol.kind == SymbolKind.TYPE_DEF && bSymbol.origin == SymbolOrigin.SOURCE) {
+                bTypeDefinitionSymbols.add((BTypeDefinitionSymbol) bSymbol);
+            }
+        }
+
+        Types types = Types.getInstance(new CompilerContext());
+        Set<String> relations = new HashSet<>();
+        int len = bTypeDefinitionSymbols.size();
+        for (int i = 0; i < len; i++) {
+            for (int j = i + 1; j < len; j++) {
+                BTypeDefinitionSymbol td1 = bTypeDefinitionSymbols.get(i);
+                BTypeDefinitionSymbol td2 = bTypeDefinitionSymbols.get(j);
+                BType t1 = td1.type;
+                BType t2 = td2.type;
+                String n1 = td1.name.getValue();
+                String n2 = td2.name.getValue();
+
+                computeSubTypeRelation(t1, t2, n1, n2, types, relations);
+                computeSubTypeRelation(t2, t1, n2, n1, types, relations);
+            }
+        }
+        return relations;
+    }
+
+    private void computeSubTypeRelation(BType sourceType, BType targetType, String sourceTypeName,
+                                        String targetTypeName, Types types, Set<String> relations) {
+        if (types.isAssignable(sourceType, targetType)) {
+            relations.add(relation(sourceTypeName, targetTypeName));
+        }
+    }
+
+    private Set<String> expectedSubTypeRelations(String filePath) throws IOException {
+        Set<String> relations = new HashSet<>();
+        List<String> lines = Files.readAllLines(resolvePath(filePath));
+        for (String line : lines) {
+            // This check can be changed based on other notations.
+            if (line.startsWith(COMMENT) && line.contains(SUBTYPE_SYMBOL)) {
```

Comment:
Can we improve the symbol validation? eg: having two spaces between the comment and the symbol will brake this

Reply:
If we add a validation we are unable to add a different comment. So I think when we add a test we have to add the correct comment.